lib/api/api.js

```diff
@@ -100,34 +101,34 @@ const api = {
                 log.trace('authentication error', { error: err });
                 return callback(err);
             }
-            if (authorizationResults) {
+            function ifAuthResults(authResults) {
                 if (apiMethod === 'objectGet') {
                     // first item checks s3:GetObject(Version) action
-                    if (!authorizationResults[0].isAllowed) {
+                    if (!authResults[0].isAllowed) {
                         log.trace('get object authorization denial from Vault');
                         return callback(errors.AccessDenied);
                     }
                     // second item checks s3:GetObject(Version)Tagging action
-                    if (!authorizationResults[1].isAllowed) {
+                    if (!authResults[1].isAllowed) {
                         log.trace('get tagging authorization denial ' +
                         'from Vault');
                         returnTagCount = false;
                     }
                 } else {
-                    for (let i = 0; i < authorizationResults.length; i++) {
-                        if (!authorizationResults[i].isAllowed) {
+                    for (let i = 0; i < authResults.length; i++) {
+                        if (!authResults[i].isAllowed) {
                             log.trace('authorization denial from Vault');
                             return callback(errors.AccessDenied);
                         }
                     }
```

Comment:
Looks like a good case for `some`.
```suggestion
                    if (authResults.some(ar => !ar.isAllowed) {
                        log.trace('authorization denial from Vault');
                        return callback(errors.AccessDenied);
                    }
```